### PR TITLE
podvm-mkosi: override kata-agent config path

### PR DESCRIPTION
**Pull request by mkulke**
fixes: #1637

~blocked-by: https://github.com/confidential-containers/guest-components/pull/429~

On a podvm using an guest-components build from the PR, `KATA_AGENT_CONFIG_PATH` is set and picked up by attestation-agent and cdh:

```bash
$ cat /proc/$(pgrep -i attest)/environ | xargs --null echo
LANG=C.UTF-8 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin SYSTEMD_SULOGIN_FORCE=1 INVOCATION_ID=ad04b8c4ff6c4f2c8617dfcfed24c183 JOURNAL_STREAM=8:3090 SYSTEMD_EXEC_PID=552 KATA_AGENT_CONFIG_PATH=/run/peerpod/agent-config.toml RUST_BACKTRACE=full OCICRYPT_KEYPROVIDER_CONFIG=/tmp/ocicrypt_config.json
```

Verified w/ a kbs deployment that the mkosi-podvm performs a remote attestation.

Once this is the guest-components PR is merged, we need to bump its revision in `versions.yaml`

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
LGTM (once the guest-components PR has merged and versions.yaml has bumped). Thanks!

File: podvm-mkosi/mkosi.skeleton/usr/lib/systemd/system/kata-agent.service.d/10-override.conf
Comment (katexochen) [inline podvm-mkosi/mkosi.skeleton/usr/lib/systemd/system/kata-agent.service.d/10-override.conf:4]: Was this removed by the original unit? Otherwise lgtm, thanks.

Comment (mkulke) [reply podvm-mkosi/mkosi.skeleton/usr/lib/systemd/system/kata-agent.service.d/10-override.conf:4]: ah, no this was not intentionally removed. but it doesn't do anything, does it? the kata-agent unit doesn't have this set.

Comment (katexochen) [reply podvm-mkosi/mkosi.skeleton/usr/lib/systemd/system/kata-agent.service.d/10-override.conf:4]: I think it had some condition at some point, but couldn't find one in current main.

**Review by bpradipt — APPROVED** [review]
/lgtm

**Pull request merged**
